parsers/EIA.py

```diff
@@ -501,52 +369,40 @@ def _fetch_series(
                 f"target_datetime must be a valid datetime - received {target_datetime}"
             )
         utc = tz.gettz("UTC")
-        if isinstance(target_datetime, datetime):
-            # eia currently only accepts utc timestamps in the form YYYYMMDDTHHZ
-            end = target_datetime.astimezone(utc).strftime("%Y%m%dT%HZ")
-            start = (target_datetime.astimezone(utc) - timedelta(days=1)).strftime(
-                "%Y%m%dT%HZ"
-            )
-        else:
-            end = None
-            start = None
-        raw_data = series.get_data(start=start, end=end)
+        eia_ts_format = "%Y-%m-%dT%HH"
+        end = target_datetime.astimezone(utc) + timedelta(hours=1)
+        start = end - timedelta(days=1)
+        url = f"{url_prefix}&api_key={API_KEY}&start={start.strftime(eia_ts_format)}&end={end.strftime(eia_ts_format)}"
     else:
-        # Get the last 24 hours available.
-        raw_data = series.last(24)
-
-    eia_error_message = raw_data.get("data", {}).get("error")
-    if eia_error_message:
-        logger.error(f"EIA error, for series_id [{series_id}]: {eia_error_message}")
-        return []
-
-    # UTC timestamp with no offset returned.
-    if not raw_data.get("series"):
-        # Series doesn't exist. Probably requesting a fuel from a region that
-        # doesn't have any capacity for that fuel type.
-        return []
+        url = f"{url_prefix}&api_key={API_KEY}&sort[0][column]=period&sort[0][direction]=desc&length=24"
 
+    s = session or Session()
+    req = s.get(url)
+    raw_data = req.json()
     return [
```

Comment:
The example parser mentions that parsers should not return empty data though so I think it might be best to raise a ParserException if no data is found/available.
> If no data can be fetched, any falsy value (None, [], False) will be
      ignored by the backend. If there is no data because the source may have
      changed or is not available, raise an Exception.

What do you think?

Reply:
I remember a similar discussion a while back but I can't seem to find it, but I think the general consensus was that if we know that data is missing it's better to have no data and let the estimation models handle the gap or refetch it in the future. But I'm not sure, might be better to ask Pierre about this one.

Reply:
Ok thanks for your reply Viktor. @pierresegonne do you have an answer to this?

Reply:
Just had a look at this specific issue. The EIA should consistently return a response with the same format. So if there is no capacity for a mode for a zone, the response will hold a `data` field which will be an empty list. To check, one can run the parser for e.g. `US-MIDW-AECI` and see that even though there is no nuclear capacity, the data field returned is just an empty list. 
So overall I think this is good as is.

For the discussion better return an empty list or fail the parser, right now our stance would be that if we know that some expected is indeed missing, we might as well crash the parser. Returning a falsy value will result in a parser failure in any case.

With the new data validation scheme, we should soon be able to ingest these faulty values though :)

Reply:
Alright, in that case it is best to add a ParserException as @VIKTORVAV99 mentioned above :)